test/post.test.js

```diff
@@ -23,3 +25,154 @@ t.test('cannot set schemaCompiler after binding', t => {
     }
   })
 })
+
+t.test('should receive application/json content-type', t => {
```

Comment:
Why do we need this test?

Reply:
I usually like to add a test case to prove that it works correctly as expected to eliminate false positives 😄 I can remove it if you think it is redundant

Reply:
Take a look at [helper.js](https://github.com/fastify/fastify/blob/master/test/helper.js), we are doing all the body related tests there ;)
You can move all of this tests in that file, so we will test every http method that allows a body.

Reply:
@delvedor done